### DISCUSSION TITLE
Filtering a Jagged Array with no fields

### DISCUSSION THREAD

**Discussion by MoAly98** [scikit-hep/awkward#1287] [Q&A]
Hi,

This might be a stupid question, but I have an awkward array that looks like:
```
arr = [ [1,2,3], [4,5], ...] 
```
and I would like to filter/mask the inner values of the array (e.g. mask elements <= 1 such that `arr -> [[None,2,3],[4,5] `). I know how to do this if I have a field, since it's a simple `arr.field > X', but in this case the array has no fields. 

Is this possible? 

Thank you very much.

**Comment by agoose77** [comment]
The masking and filtering features in NumPy / Awkward are actually orthogonal (unrelated) to the fields feature. 

If you have an array e.g.
```python
x = ak.Array([
    [1,2,3],
    [4, 5],   
    [6, 7, 8, 9]
])
```
then to mask _out_ the elements `<=1`, you can just use `.mask`, e.g.
```python
y = x.mask[x > 1]
```

The [top-level function `ak.mask`](https://awkward-array.readthedocs.io/en/latest/_auto/ak.mask.html?highlight=mask#ak.mask), unlike `ak.Array.mask` actually accepts a `valid_when` argument that you can use to invert the meaning of the boolean mask array (i.e. if you want `True` to mean "set this value to `None`").

For more information on filtering and masking, I recommend reading [the docs](https://awkward-array.readthedocs.io/en/latest/_auto/ak.Array.html?highlight=mask#filtering) on the topic :)
